### ISSUE TITLE
Star Ocean 5 can only be started once per Windows boot due to MEM: Access violation

### ISSUE THREAD

**Issue by Metafalica** [RPCS3/rpcs3#6825]
Just like the title says this game can be started only once per Windows boot. Any other attempts to start it ends with **Thread (main_thread) [0x0051407c]} MEM: Access violation reading location**.

**How to reproduce:** Start the game once, choose new game, see beginning cut scene and just try to play until emu freeze with **Thread (main_thread) [0x0051407c]} MEM: Access violation reading location**. It will freeze soone enough... After this no matter how many times you try to start it, emu will always freeze with same error before displaying any game start screens.
If you restart emulation without waiting for first freeze everything is OK. It will not freeze at next game boot.

RPCS3 v0.0.7-8968
Intel(R) Core(TM) i7 CPU 880 @ 3.07GHz | 8 Threads | 7.96 GiB RAM | TSC: Bad
Radeon HD 5750
Core:
  PPU Decoder: Recompiler (LLVM)
  PPU Threads: 2
  PPU Debug: false
  Save LLVM logs: false
  Use LLVM CPU: ""
  Max LLVM Compile Threads: 0
  Enable thread scheduler: true
  Set DAZ and FTZ: false
  SPU Decoder: Recompiler (LLVM)
  Lower SPU thread priority: false
  SPU Debug: false
  Preferred SPU Threads: 0
  SPU delay penalty: 3
  SPU loop detection: true
  Max SPURS Threads: 6
  SPU Block Size: Safe
  Accurate GETLLAR: false
  Accurate PUTLLUC: false
  SPU Verification: true
  SPU Cache: true
  SPU Profiler: false
  Enable TSX: Enabled
  Accurate xfloat: false
  Approximate xfloat: true
  Debug Console Mode: false
  Lib Loader: Load liblv2.sprx only
  Hook static functions: false
  Load libraries:
    []
  HLE lwmutex: false
  Clocks scale: 100
  Sleep Timers Accuracy: Usleep Only
VFS:
  $(EmulatorDir): ""
  /dev_hdd0/: $(EmulatorDir)dev_hdd0/
  /dev_hdd1/: $(EmulatorDir)dev_hdd1/
  /dev_flash/: ""
  /dev_usb000/: $(EmulatorDir)dev_usb000/
  /dev_bdvd/: ""
  /app_home/: ""
  Enable /host_root/: false
  Initialize Directories: true
  Limit disk cache size: false
  Disk cache maximum size (MB): 5120
Video:
  Renderer: OpenGL
  Resolution: 1280x720
  Aspect ratio: 16:9
  Frame limit: Auto
  MSAA: Auto
  Write Color Buffers: false
  Write Depth Buffer: false
  Read Color Buffers: false
  Read Depth Buffer: false
  Log shader programs: false
  VSync: false
  Debug output: false
  Debug overlay: false
  Use Legacy OpenGL Buffers: false
  Use GPU texture scaling: false
  Stretch To Display Area: false
  Force High Precision Z buffer: false
  Strict Rendering Mode: false
  Disable ZCull Occlusion Queries: false
  Disable Vertex Cache: false
  Disable FIFO Reordering: false
  Enable Frame Skip: false
  Force CPU Blit: false
  Disable On-Disk Shader Cache: false
  Disable Vulkan Memory Allocator: false
  Use full RGB output range: true
  Disable Asynchronous Shader Compiler: true
  Strict Texture Flushing: false
  Disable native float16 support: false
  Multithreaded RSX: false
  Consecutive Frames To Draw: 1
  Consecutive Frames To Skip: 1
  Resolution Scale: 100
  Anisotropic Filter Override: 0
  Minimum Scalable Dimension: 16
  Driver Recovery Timeout: 1000000
  Vblank Rate: 60
  D3D12:
    Adapter: ""
  Vulkan:
    Adapter: ""
    Force FIFO present mode: false
    Force primitive restart flag: false
  Performance Overlay:
    Enabled: false
    Detail level: Medium
    Metrics update interval (ms): 350
    Font size (px): 10
    Position: Top Left
    Font: n023055ms.ttf
    Horizontal Margin (px): 50
    Vertical Margin (px): 50
    Center Horizontally: false
    Center Vertically: false
    Opacity (%): 70
    Body Color (hex): "#FFE138FF"
    Body Background (hex): "#002339FF"
    Title Color (hex): "#F26C24FF"
    Title Background (hex): "#00000000"
  Shader Compilation Hint:
    Position X (px): 20
    Position Y (px): 690
  Shader Loading Dialog:
    Allow custom background: true
    Darkening effect strength: 30
    Blur effect strength: 0
Audio:
  Renderer: XAudio2
  Dump to file: false
  Convert to 16 bit: false
  Downmix to Stereo: true
  Start Threshold: 1
  Master Volume: 100
  Enable Buffering: true
  Desired Audio Buffer Duration: 100
  Sampling Period Multiplier: 100
  Enable Time Stretching: false
  Time Stretching Threshold: 75
  Microphone Type: "Null"
  Microphone Devices: "@@@@@@@@@@@@"
Input/Output:
  Keyboard: "Null"
  Mouse: Basic
  Pad: Keyboard
  Camera: "Null"
  Camera type: Unknown
  Move: "Null"
System:
  Language: English (US)
  Keyboard Type: English keyboard (US standard)
  Enter button assignment: Enter with cross
Net:
  Connection status: Disconnected
  IP address: 192.168.1.1
Miscellaneous:
  Automatically start games after boot: true
  Exit RPCS3 when process finishes: false
  Start games in fullscreen mode: false
  Prevent display sleep while running games: true
  Show FPS counter in window title: true
  Show trophy popups: true
  Show shader compilation hint: true
  Use native user interface: true
  GDB Server: 127.0.0.1:2345

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3754234/RPCS3.log.gz)

**Comment by AniLeo** [comment]
Probably duplicate of #3872

**Comment by Metafalica** [comment]
Maybe, but not sure... That ticket is very old, a lot of thing changed and there is no explanation when their freeze happens.

**Comment by rapierx** [comment]
If you use RAMMap after closing RPCS3 to empty modified list page, standby list and priority 0 standby list (flushing RPCS3 from memory) you can start the game again.

**Comment by Metafalica** [comment]
Yuck, I doubt this is how we need to use rpcs3.

**Comment by Metafalica** [comment]
Nothing changed in current latest build. I know it's hard to obtain SO5. I can give it for testing if that's the problem behind many years of ignore.

**Comment by Metafalica** [comment]
Fresh log with access violation at v0.0.18-12750.
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/7179242/RPCS3.log)

**Comment by Megamouse** [comment]
Please re-test

**Comment by Metafalica** [comment]
No changes in v0.0.20-13198.

**Comment by ENjoyBlue2021** [comment]
I never got this game to work, even the first attempt fails for me.
Always the **RPCS3: PPU[0x1000000] Thread (main_thread) [0x00016700]: VM: Access violation reading location 0x0 (unmapped memory)** message.
Made a switch from Win10 to Ubuntu and it still crashes with the same message.
Tried various settings but nothing seems to work. Only game for me which I cant get ingame at all.

**Comment by illusion0001** [comment]
Try with https://github.com/RPCS3/rpcs3/pull/12172

**Comment by Metafalica** [comment]
No changes in current latest build

> Try with #12172

Sorry, I don't understand where to download this build. Give a link for Windows, please.

**Comment by Darkhost1999** [comment]
> Sorry, I don't understand where to download this build. Give a link for Windows, please.

https://cirrus-ci.com/task/4543090907152384
Just click -> artifact -> download the 7z.
Take the rpcs3.exe and extract it from the zip. Rename it to anything other than rpcs3.exe and add it to your existing rpcs3 folder. But the .exe you renamed and try the game. Remember to share logs and if it worked or not.

**Comment by VIRGINKLM** [comment]
I wonder why RPCS3 "hates" that game so badly. It rejects it so hard like a body rejecting a bad transplant. I've been keeping an eye on progress on that game for years now but there's not a lot of progress into what makes RPCS3 crash so extensively on that game. Rather interesting mistery.

**Comment by Metafalica** [comment]
I can't believe it, but seems like game became playable now (since commit that reject trash shaders compilation). Emu was barely able to run this game up to training scene, because of "MEM: Access violation" junk, but now I successfully played far beyond that and not a single crash happened.
I use 0.0.26-14689 and I set *RSX FIFO Accuracy* to *Atomic* on *Advanced* tab in settings.

**Comment by AniLeo** [comment]
Reported to still not boot on current builds even with Atomic FIFO, we need a new log file of the issue

**Comment by Metafalica** [comment]
It boots totally fine for a long time on different constantly updating emu versions. As long as it first game boot after booting Windows...
![game_ok](https://github.com/RPCS3/rpcs3/assets/5286295/a320d3bd-8c4e-4afa-a79d-4280e8d90c74)
Log when it boots fine:
[RPCS3_first_boot.log](https://github.com/RPCS3/rpcs3/files/13327676/RPCS3_first_boot.log)

And this what happens when I try to boot it second time (without restarting Windows):
![game_not_ok](https://github.com/RPCS3/rpcs3/assets/5286295/5b00fd5b-3be1-4d01-93fa-94feba7dc74c)
Log when emu freeze:
[RPCS3_second_boot.log](https://github.com/RPCS3/rpcs3/files/13327681/RPCS3_second_boot.log)

This ticket title edit was wrong. Is was displaying exact state of this game.

**Comment by AniLeo** [comment]
Thanks, I restored the original title as we had several reports of it not working at all - someone could test this on Linux by the way to confirm if it's an OS specific bug or not

**Comment by Metafalica** [comment]
Just tested native rpcs3 AppImage on Steam Deck desktop mode (ArchLinux based). Game boots first time and freeze all later times exactly like on Windows. I can't even imagine how program must be coded to achieve same behavior when if you decide to repeat this intentionally.

**Comment by FlexBy420** [comment]
Using the PPU interpreter allows to get ingame without crash most of the times.

**Comment by elad335** [comment]
Retest with master

**Comment by Metafalica** [comment]
Tested v0.0.30-15945-3513f6d8. No changes (of course). Still freeze on second time.

**Comment by digant73** [comment]
@Metafalica is that still an issue?

**Comment by Metafalica** [comment]
Tested v0.0.38-18280. No changes (of course, until some dev run emu under debugger to catch it).

**Comment by elad335** [comment]
@Metafalica  Can you upload an updated log?

**Comment by Metafalica** [comment]
[RPCS3.zip](https://github.com/user-attachments/files/23290884/RPCS3.zip)
This is second start with fast MEM: Access violation crash.

**Comment by ENjoyBlue2021** [comment]
This is still a issue every second time you open the game.
Im on latest kubuntu.

Maybe this is useful for somebody stumbling on this:
A workaround I did to get it to work:
sync
sudo sh -c 'echo 3 > /proc/sys/vm/drop_caches'

Before starting rpcs3. This cleans up RAM.
Weirdly enough even a system reset did not work without doing the above 2 commands.

**Comment by LeviCroker** [comment]
On steam deck (Linux) it runs without issue until the first battle against bandits attacking the village. The story doesn't advance after defeating the first two bandits. It just remains stuck in battle with no enemies. 

My setup is LLVM Recompilers with Vulkan GPU. The game crashes when I make changes to any setting.

My logs: 

[RPCS3.log](https://github.com/user-attachments/files/27098101/RPCS3.log)

<img width="861" height="484" alt="Image" src="https://github.com/user-attachments/assets/c375bb80-8838-47f3-9d99-0085d4c45c8a" />

**Comment by digant73** [comment]
@Metafalica please an update on current build

**Comment by Metafalica** [comment]
Tested rpcs3-v0.0.41-19511-87608865_win64, no changes.

**Comment by Metafalica** [comment]
No changes about being able to run it only once per OS boot, but changes about performance and visual bugs are clearly visible.
I had stable 30 FPS before, now it run with frequent drops to 10 FPS (and I have Intel Xeon CPU with 14 physical cores!!!), also there is graphical bug now when blue rectangle displayed instead of character phrase that usually displayed without any rectangle.

<img width="1282" height="752" alt="Image" src="https://github.com/user-attachments/assets/e7407e8c-10a0-4d5c-bf38-c937eb56c193" />
